pkg/infra/httpclient/httpclientprovider/grafana_request_id_header_middleware.go

```diff
@@ -0,0 +1,59 @@
+package httpclientprovider
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"net/http"
+
+	"github.com/google/uuid"
+
+	sdkhttpclient "github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/clientmiddleware"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+const GrafanaRequestIDHeaderMiddlewareName = "grafana-request-id-header-middleware"
+
+func GrafanaRequestIDHeaderMiddleware(cfg *setting.Cfg) sdkhttpclient.Middleware {
+	return sdkhttpclient.NamedMiddlewareFunc(GrafanaRequestIDHeaderMiddlewareName, func(opts sdkhttpclient.Options, next http.RoundTripper) http.RoundTripper {
+		return sdkhttpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			foundMatch := false
+			for _, allowedURL := range cfg.IPRangeACAllowedURLs {
+				// Only look at the scheme and host, ignore the path
+				if allowedURL.Host == req.URL.Host && allowedURL.Scheme == req.URL.Scheme {
+					foundMatch = true
+					break
+				}
+			}
```

Comment:
Nit. Feels like this logic could be shared between client middleware and http client middleware?